Add Recipe for Using Tailwind to Style Rendered Markdown

VoxelMC opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### Description (required)
This PR aims to add an official recipe to the Astro documentation which describes how you can use Tailwind's Typography plugin to style rendered Markdown, such as content collection content.

New Recipe: [/en/recipes/tailwind-rendered-markdown/](https://deploy-preview-5272--astro-docs-2.netlify.app/en/recipes/tailwind-rendered-markdown/)

#### Related issues & labels (optional)

- Closes #4367
- Suggested label: add new content, recipe

sarah11918 commented:
Looking forward to digging into this one over the weekend!!  :partying_face:

jacobthesheep commented:
I think we should follow the ESM syntax for the `tailwind.config` file

jacobthesheep commented on src/content/docs/en/recipes/tailwind-rendered-markdown.mdx:
```diff
@@ -0,0 +1,110 @@
+// tailwind.config.js
+/** @type {import('tailwindcss').Config} */
+
+module.exports = {
```
Replace with `export default defineConfig = {`

VoxelMC replied:
I do believe that the proper ESM syntax for this export would be to just export an object like so:
```diff
/** @type {import('tailwindcss').Config} */
- module.exports = {
+ export default {
  // ...
}
```

I could definitely fix that, since the Tailwind plugin for Astro generates a config file using ESM like I show above :)

sarah11918 replied:
I included this in my editing pass. Please correct if needed!

jsparkdev commented on src/content/docs/en/recipes/tailwind-rendered-markdown.mdx:
```diff
@@ -0,0 +1,110 @@
+
+## Resources
+
+[Tailwind Typography Documentation](https://tailwindcss.com/docs/typography-plugin)
```
```suggestion
- [Tailwind Typography Documentation](https://tailwindcss.com/docs/typography-plugin)
```

Match resource list format with other documents

sarah11918 commented:
@VoxelMC Thank you for this great recipe! It is well thought-out and well-written.

I took an editing pass, see what you think! We can work from it and make any changes in this state.

Some notable choices:
- I removed the external StackBlitz link because that's not very future-proof. The recipe could change here (as it already did!) or anything could happen to that repo...
- I changed the kind of handwavy "make your markdown shine!" to actually describe what the feature description is: being able to use Tailwind's utility classes on your Markdown. This is also better if people are searching the docs as it gives very specific keywords that they can find this recipe from.
- I moved the section on element modifiers to the bottom, so it's not in the middle of instructions (since it's not a thing to do) and linked to it. Honestly, in some moods, I would have just linked externally and removed that section entirely saying it's not our job to document **how Tailwind works**, and they already have documentation on this topic, but it was short, and well-written so I kept it. 😄 
- I was way more opinionated throughout (e.g. instead of *if* you're using Content collections... Nope! You use content collections now!) so that the instructions and examples can be cleaner, made with certain assumptions. People are free to modify at will!
- Some of my instructions were a little more explicit (referencing actual elements etc.) to draw the attention to the key points of the code samples that were doing the heavy lifting to get our desired affect.

I know that list looks like a lot, and some very specific minor changes, but when it was this good to begin with, it's easy to spot the areas that just need that little tweak to **make your recipe shine!** 🌟

sarah11918 commented on src/content/docs/en/recipes/tailwind-rendered-markdown.mdx:
```diff
@@ -0,0 +1,108 @@
+An Astro project that:  
+
+	- has [Astro's Tailwind integration](/en/guides/integrations-guide/tailwind/) installed.  
+	- uses Astro's [content collections](/en/guides/integrations-guide/content-collections/).
```
```suggestion
	- uses Astro's [content collections](/en/guides/content-collections/).
```

VoxelMC commented on src/content/docs/en/recipes/tailwind-rendered-markdown.mdx:
```diff
@@ -0,0 +1,108 @@
+// tailwind.config.js
+/** @type {import('tailwindcss').Config} */
+
+export default = {
```
```suggestion
export default {
```

VoxelMC commented:
> @VoxelMC Thank you for this great recipe! It is well thought-out and well-written.
> 
> I took an editing pass, see what you think! We can work from it and make any changes in this state.
> 
> Some notable choices:
> 
> * I removed the external StackBlitz link because that's not very future-proof. The recipe could change here (as it already did!) or anything could happen to that repo...
> * I changed the kind of handwavy "make your markdown shine!" to actually describe what the feature description is: being able to use Tailwind's utility classes on your Markdown. This is also better if people are searching the docs as it gives very specific keywords that they can find this recipe from.
> * I moved the section on element modifiers to the bottom, so it's not in the middle of instructions (since it's not a thing to do) and linked to it. Honestly, in some moods, I would have just linked externally and removed that section entirely saying it's not our job to document **how Tailwind works**, and they already have documentation on this topic, but it was short, and well-written so I kept it. 😄
> * I was way more opinionated throughout (e.g. instead of _if_ you're using Content collections... Nope! You use content collections now!) so that the instructions and examples can be cleaner, made with certain assumptions. People are free to modify at will!
> * Some of my instructions were a little more explicit (referencing actual elements etc.) to draw the attention to the key points of the code samples that were doing the heavy lifting to get our desired affect.
> 
> I know that list looks like a lot, and some very specific minor changes, but when it was this good to begin with, it's easy to spot the areas that just need that little tweak to **make your recipe shine!** 🌟

Thank you for your edits, inputs, and inspiration! I will do a readthrough and double check that all of the code works, make some edits, and let you know when it's ready for another look.

Also, I will take the advice to any future recipes I make! It all makes total sense, in hindsight :P

sarah11918 commented on src/content/docs/en/recipes/tailwind-rendered-markdown.mdx:
```diff
@@ -0,0 +1,106 @@
+
+## Recipe
+
+1. Create a `<Prose /> ` component to provide a wrapping `<div>` with a `<slot />` for your rendered Markdown. Add the style class `prose` alongside any desired **[Tailwind element modifiers](#how-to-use-element-modifiers)** in the parent element.   
```
```suggestion
1. Create a `<Prose /> ` component to provide a wrapping `<div>` with a `<slot />` for your rendered Markdown. Add the style class `prose` alongside any desired [Tailwind element modifiers](https://tailwindcss.com/docs/typography-plugin#element-modifiers) in the parent element.   
```

sarah11918 commented on src/content/docs/en/recipes/tailwind-rendered-markdown.mdx:
````diff
@@ -0,0 +1,106 @@
+      prose-[element]:class-to-apply
+      ``` 
+
+    For example, `prose-h1:font-bold` gives all `<h1>` tags the `font-bold` Tailwind class.
````
```suggestion
    For example, `prose-h1:font-bold` gives all `<h1>` tags the `font-bold` Tailwind class.
    :::
```
Not sure why this didn't break withhout the closing part?  :sweat_smile:

sarah11918 reviewed (approved):
Really happy with how this turned out! Thank you!
